crates/core/component/dex/src/batch_swap_output_data.rs

```diff
@@ -142,9 +155,9 @@ pub struct BatchSwapOutputDataVar {
     pub lambda_2: U128x128Var,
     pub unfilled_1: U128x128Var,
     pub unfilled_2: U128x128Var,
-    pub height: FqVar,
     pub trading_pair: TradingPairVar,
-    pub epoch_starting_height: FqVar,
+    pub epoch: FqVar,
+    pub block: FqVar,
```

Comment:
What's the rationale behind making these an `FqVar` rather than an alternate representation?

Also, could we give them different names to clarify that they're not standalone values, but components of a `Position`? Like
```
pub position_epoch_component: FqVar,
pub position_block_component: FqVar,
```
or some other name? I'm just worried that calling it `block` could suggest it's the block height, which it isn't.

Reply:
Changed to store a PositionVar now

Reply:
No I meant, what's the rationale for the representation? I'm not opposed to using an FqVar, I'm just wondering how we decide one or the other. Can we write up a brief rationale for why we picked one or the other?

Reply:
A `PositionVar` has a position stored internally as both an `FqVar` and a series of 48 bit constraints - 16 bits each for the epoch, block, and commitment segment of the position. Since we need to compare the epoch and block from the position from the state commitment proof with the epoch and block from the BSOD, and we're already bit constraining to make two FqVars for the epoch and block in both cases, another (I think cheaper in constraints) option is to compare as bits directly. Comparing as `FqVar`s also works though.

Reply:
Hmm, in that case wouldn't making the two prefix components FqVars be cheaper? Since we already have the bit-constrained components of the actual position, we can check that they're equal to the provided prefix components without needing to bit-constrain the prefix components, right?

Reply:
yep, you're right. That seems like the way to go to save the additional bit constraints

Reply:
Not sure that comparing bits is cheaper in constraints: checking that (x0 + 2 * x1 + ...) =? (y0 + 2 * y1 + ...) can be done in a single R1CS constraint. Checking x0 =? y0, x1 =? y1 would require several

Reply:
Yeah, that seems like a good reason to use FqVars as in the original design, I think it's clearer than using a Position because it means we don't add any instance variables that don't have meaning but could be (mis)used in the circuit.